projects/plugins/crm/src/automation/data-transformers/class-data-transformer-object-to-tag-list.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**
+ * CRM Object to CRM Tag List Transformer class.
+ *
+ * @package automattic/jetpack-crm
+ */
+
+namespace Automattic\Jetpack\CRM\Automation\Data_Transformers;
+
+use Automattic\Jetpack\CRM\Automation\Data_Transformer_Exception;
+use Automattic\Jetpack\CRM\Automation\Data_Types\Data_Type;
+use Automattic\Jetpack\CRM\Automation\Data_Types\Tag_List_Data;
+
+/**
+ * CRM Object to CRM Tag List Transformer class.
+ *
+ * @since $$next-version$$
+ */
+class Data_Transformer_Object_To_Tag_List extends Data_Transformer_Base {
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public static function get_slug(): string {
+		return 'object_to_tag_list';
+	}
+
+	/**
+	 * Get the slug name of the Data Type.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param Data_Type $data The object data type we want to get the slug from.
+	 * @return string The slug of the Data Type.
+	 *
+	 * @throws Data_Transformer_Exception If the object type cannot be transformed to a tag list.
+	 */
+	public static function get_from( Data_Type $data ): string {
+
+		switch ( $data ) {
+			case ( $data instanceof Company ):
+				return Company_Data::class;
+			case ( $data instanceof Contact ):
+				return Contact_Data::class;
+			case ( $data instanceof Invoice ):
+				return Invoice_Data::class;
+			case ( $data instanceof Transaction ):
+				return Transaction_Data::class;
+			case ( $data instanceof Quote ):
+				return Quote_Data::class;
+			case ( $data instanceof Task ):
+				return Task_Data::class;
+			default:
+				throw new Data_Transformer_Exception(
+					'Object type cannot be transformed to tag list.',
+					Data_Transformer_Exception::TRANSFORM_IS_NOT_SUPPORTED
+				);
+		}
+	}
```

Comment:
I'm refactoring this, to use a new generic Data_Type to identify JPCRM Data_Type. It's a simple interface called Entity_Data, implemented by the JPCRM Data_Type. The transformer just has to accept Entity_Data::class.